From 162be406957690e7e3fdbb966549df2a2ad89a88 Mon Sep 17 00:00:00 2001
From: Amruthavani <amruthavani@mazhar-desktop.(none)>
Date: Sat, 22 Sep 2012 13:35:12 +0530
Subject: [PORT FROM MAIN]Audio-Offload: Resume playback on BT during call

BZ: 57562

This patch will fix
- Resume playback on BT during call
- Resume playback after enabling effects with long pause

Change-Id: I83f7c8b4295092d1a380f325980358dc418581a2
Orig-Change-Id: Ic52dd60778d877ef752e92767ad16f3494e3c3f9
Signed-off-by: samaga Prasanna Krishna <samaga.krishna@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |   16 ++++++++++++++--
 1 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 8bcced4..44ad267 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -968,8 +968,11 @@ status_t AwesomePlayer::play() {
     //  Before play, we should query audio flinger to see if any effect is enabled.
     //  if (effect is enabled) we should do another prepare w/ IA SW decoding
     if (mOffload && (isAudioEffectEnabled() ||
-        AudioSystem::getDeviceConnectionState(AUDIO_DEVICE_OUT_AUX_DIGITAL, "") == AUDIO_POLICY_DEVICE_STATE_AVAILABLE)) {
-        ALOGV("Offload and effects are enabled or HDMI");
+        (AudioSystem::getDeviceConnectionState(AUDIO_DEVICE_OUT_AUX_DIGITAL, "")
+         == AUDIO_POLICY_DEVICE_STATE_AVAILABLE) ||
+        (AudioSystem::getDeviceConnectionState(AUDIO_DEVICE_OUT_BLUETOOTH_A2DP, "")
+         == AUDIO_POLICY_DEVICE_STATE_AVAILABLE))) {
+        ALOGV("Offload and effects are enabled or HDMI or BT connected");
         mAudioOffloadTearDownEventPending = true;
         modifyFlags(PLAYING, CLEAR);
         onAudioOffloadTearDownEvent();
@@ -3148,6 +3151,15 @@ status_t AwesomePlayer::offloadResume() {
 
     seekTo_l(stats.mPositionUs);
     mFlags = stats.mFlags & (AUTO_LOOPING | LOOPING | AT_EOS);
+
+    if (mOffloadTearDownForPause && (isAudioEffectEnabled() ||
+        (AudioSystem::getDeviceConnectionState(AUDIO_DEVICE_OUT_AUX_DIGITAL, "")
+         == AUDIO_POLICY_DEVICE_STATE_AVAILABLE) ||
+        (AudioSystem::getDeviceConnectionState(AUDIO_DEVICE_OUT_BLUETOOTH_A2DP, "")
+         == AUDIO_POLICY_DEVICE_STATE_AVAILABLE))) {
+        mOffload = false;
+    }
+
     play_l();
     mOffloadTearDownForPause = false;
     // Update the flag
-- 
1.7.4.1

